### DISCUSSION TITLE
cmake build error

### DISCUSSION THREAD

**Discussion by wzzlearning** [obsproject/obs-studio#11136] [Development Questions]
D:\study\obs-studio>cmake --preset windows-x64
Preset CMake variables:

  ENABLE_BROWSER:BOOL="TRUE"
  ENABLE_CCACHE:BOOL="FALSE"
  GPU_PRIORITY_VAL:STRING=""
  OBS_CMAKE_VERSION:STRING="3.0.0"
  RESTREAM_CLIENTID:STRING=""
  RESTREAM_HASH:STRING=""
  TWITCH_CLIENTID:STRING=""
  TWITCH_HASH:STRING=""
  VIRTUALCAM_GUID:STRING="A3FCE0F5-3493-419F-958A-ABA1250EC20B"
  YOUTUBE_CLIENTID:STRING=""
  YOUTUBE_CLIENTID_HASH:STRING=""
  YOUTUBE_SECRET:STRING=""
  YOUTUBE_SECRET_HASH:STRING=""

-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.22621.
-- Setting up Pre-Built obs-deps (x64)
-- Setting up Pre-Built obs-deps (x64) - skipped
-- Setting up Pre-Built Qt6 (x64)
-- Setting up Pre-Built Qt6 (x64) - skipped
-- Setting up Chromium Embedded Framework (x64)
-- Setting up Chromium Embedded Framework (x64) - skipped
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.22621.
-- Setting up Pre-Built obs-deps (x86)
-- Setting up Pre-Built obs-deps (x86) - skipped
CMake Warning (dev) at cmake/finders/FindDetours.cmake:81 (message):
  Failed to find detours version.
Call Stack (most recent call first):
  plugins/win-capture/graphics-hook/CMakeLists.txt:3 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring done (0.3s)
CMake Error at plugins/win-dshow/virtualcam-module/cmake/32bit-build.cmake:47 (target_link_libraries):
  Cannot find source file:

    D:/study/obs-studio/plugins/win-dshow/libdshowcapture/dshowcapture.hpp
Call Stack (most recent call first):
  cmake/32bit/projects.cmake:15 (include)
  plugins/win-dshow/virtualcam-module/CMakeLists.txt:84 (legacy_check)

CMake Error: CMake can not determine linker language for target: obs-virtualcam-module
-- Generating done (0.3s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
CMake Error at cmake/windows/defaults.cmake:28 (execute_process):
  execute_process failed command indexes:

    1: "Child return code: 1"

Call Stack (most recent call first):
  CMakeLists.txt:14 (include)

-- Configuring incomplete, errors occurred!

what should I do？

**Comment by WizardCM** [comment]
Make sure you cloned the repository recursively. If not, you may need to init and download the submodules.
